### ISSUE TITLE
[BUG] F12 - Model/Asset Type Association's logic has been broken

### ISSUE THREAD

**Issue by JayChouteau** [rdsaliba/notorious-eng#292]
**Describe the bug**
When trying to save a model association for a specific asset type, the association doesn't get saved properly. The feature doesn't work anymore.

**To Reproduce**
1. Go to asset type list
2. Click on any asset type in the list
3. Click on the model tab
4. Select a model that is not currently selected (borders aren't red)
5. Click on Save
6. Click back on the same asset type from the list
7. On the information tab, read the associated RUL model text box -> see error
8. On the model tab, check which thumbnail has red borders -> see error

**Expected behavior**
The selection should be properly saved to be displayed in the UI

**Screenshots**

**Desktop (please complete the following information):**
 - OS: any OS

**Additional context**
The previous logic of saving an association between the model and the asset type was based on the fact that the trained_model table in the database would only hold 1 trained model per asset type. This meant that selecting a model and saving the selection would change the model ID in the trained model table for the asset type in the database and change the retrain tag to true. However, since issues #222, #276, and #270, the trained_model table holds multiple model entries for 1 asset type.  Since then, another attribute was added to the table to keep track of which entries are meant for evaluation and which entries are meant for the application (status_id). Most likely, the logic of the association will have to be updated to take into consideration these new changes.

TL:DR -> The logic has been broken and needs to be fixed.

**Comment by JayChouteau** [comment]
Observations:
- When a model was selected and save was pressed, the different functions attached to the save button had the desired behavior in the back-end (database). For example, I tried to associated Asset Type (ID=1) with Model (ID=4):
1. The asset_type would be associated with the new selection
2. The asset type would be set to retrain = true
![BUG_Trained_model_Updated](https://user-images.githubusercontent.com/47158909/111511484-2b9a3c00-8725-11eb-84ba-625d8565715f.PNG)
3. All assets of that asset type would have their updated tag set to true:
![Bug_Asset_updated](https://user-images.githubusercontent.com/47158909/111511577-45d41a00-8725-11eb-8b83-44f2041467b4.PNG)
4. After pressing save, the screen would switch back to the asset type list
5. However, when clicking on the same asset type from the list, it seemed like the model selection change wasn't saved.
- On the info tab, the associated RUL model is not updated
- On the model tab, the thumbnail with red borders wasn't the one saved
- All assets info page of that asset type didn't show the saved model as their RUL model

**Comment by JayChouteau** [comment]
SOLUTION:
The queries to get the names and IDs of the associated models in the trained table weren't updated with the trained model table design change. The queries weren't checking the status_id, meaning it would check the first entry found, whether the model was meant for evaluation or live. I added a status_id check for LIVE (id=1) for the model getters in ModelDAOImpl class. The 2 methods were the following:
- getModelNameAssociatedWithAssetType()
- getModelIDAssociatedWithAssetType()

After changing that, the expected display was correct. I tested it again by choosing SMOReg for asset type (id-1):
- On the asset type information page, the associated RUL model was correctly displayed:
![BUG_Asset_Type_Info_Updated](https://user-images.githubusercontent.com/47158909/111512727-710b3900-8726-11eb-9a3c-622581e26934.PNG)
- on the asset type info - model tab page, the red-bordered thumbnail was the correct one
![BUG_Asset_Type_Info_Model_Selection_Updated](https://user-images.githubusercontent.com/47158909/111512805-86806300-8726-11eb-8a6c-a97d3bd58fc0.PNG)
- All assets of that asset type were properly changed to display the correct RUL model applied to them:
![BUG_Asset_info_updated_with_model](https://user-images.githubusercontent.com/47158909/111512875-98fa9c80-8726-11eb-8019-ff892caa0ba6.PNG)
